composer/algorithms/gyro_dropout/README.md

```diff
@@ -0,0 +1,79 @@
+# 🍰 Gyro Dropout
+
+
+[\[How to Use\]](#how-to-use) - [\[Suggested Hyperparameters\]](#suggested-hyperparameters) - [\[Technical Details\]](#technical-details) - [\[Attribution\]](#attribution)
+
+ `Computer Vision`
+
+Gyro Dropout replaces implementations of `torch.nn.Dropout`. The Gyro Dropout provides increased accuracy compared with dropout.
+
+Gyro dropout is a variant of dropout that improves the efficiency of training neural networks.
+Instead of randomly dropping out neurons in every training iteration, gyro dropout pre-selects and trains a fixed
+number of subnetworks. 'Sigma' is the number of total pre-selected subnetworks and 'Tau' is the number of concurrently scheduled subnetworks in an iteration.
+
+## How to Use
+
+### Functional Interface
+
+```python
+# Apply surgery on the model to swap-in the Gyro Dropout using the Composer functional API
+
+import composer.functional as cf
+
+def training_loop(model, train_loader):
+    cf.apply_gyro_dropout(
+        model,
+        iters_per_epoch = 196,
+        max_epoch = 100,
+        p = 0.5
+        sigma = 256,
+        tau = 16,
+    )
+
+    opt = torch.optim.Adam(model.parameters())
+    loss_fn = F.cross_entropy
+    model.train()
+
+    for X, y in train_loader:
+        y_hat = model(X)
+        loss = loss_fn(y_hat, y)
+        loss.backward()
+        opt.step()
+        opt.zero_grad()
+```
+
+### Composer Trainer
+
+```python
+from composer.trainer import Trainer
+from composer.algorithms import GyroDropout
+
+trainer = Trainer(model=model,
+                  train_dataloader=train_dataloader,
+                  eval_dataloader=eval_dataloader,
+                  max_duration='100ep',
+                  algorithms=[GyroDropout(p=0.5,
+                                          sigma=256,
+                                          tau=16)])
+
+trainer.fit()
+```
```

Comment:
Before this, the following snippet needs to be inserted:

<!--pytest.mark.gpu-->
\<!--
\```python
from torch.utils.data import DataLoader
from tests.common import RandomClassificationDataset, SimpleModel

model = SimpleModel()
train_dataloader = DataLoader(RandomClassificationDataset())
eval_dataloader = DataLoader(RandomClassificationDataset())
\```
\-->
<!--pytest-codeblocks:cont-->

to setup the doctest. This is done in other algorithm READMEs that you can take a look at. This is so these code snippets can successfully be autotested (its one of the current test failures)

Reply:
example from `label_smoothing`: 
<img width="652" alt="image" src="https://user-images.githubusercontent.com/17102158/205365199-6c89f16b-8897-4f43-8356-9ef22abecab9.png">